src/js/player.js

```diff
@@ -3749,59 +3750,153 @@ class Player extends Component {
   }
 
   /**
-   * Set layout breakpoints on the player.
+   * Removes the current breakpoint.
    *
-   * @param {boolean|Object} breakpoints
-   *        A boolean indicating whether breakpoints should or should not be
-   *        used - or an object describing custom max widths for the supported
-   *        breakpoints.
+   * @private
    */
-  setBreakpoints(breakpoints) {
-    const hadBps = Boolean(this.breakpoints_);
+  removeCurrentBreakpoint_() {
+    const className = this.currentBreakpointClass();
 
-    if (!breakpoints) {
-      this.breakpoint_ = null;
-      this.breakpoints_ = null;
-      this.off('playerresize', this.updateCurrentBreakpoint_);
-      return;
+    this.breakpoint_ = '';
+
+    if (className) {
+      this.removeClass(className);
     }
+  }
+
+  /**
+   * Set breakpoints on the player.
+   *
+   * Calling this method will remove any previous custom breakpoints and start
+   * from the defaults again.
+   *
+   * @param {Object} [breakpoints]
+   *        An object describing custom breakpoints. Leave this option off to
+   *        set default breakpoints.
+   *
+   * @param {number} [breakpoints.tiny]
+   *        The maximum width for the "vjs-layout-tiny" class.
+   *
+   * @param {number} [breakpoints.xsmall]
+   *        The maximum width for the "vjs-layout-x-small" class.
+   *
+   * @param {number} [breakpoints.small]
+   *        The maximum width for the "vjs-layout-small" class.
+   *
+   * @param {number} [breakpoints.medium]
+   *        The maximum width for the "vjs-layout-medium" class.
+   *
+   * @param {number} [breakpoints.large]
+   *        The maximum width for the "vjs-layout-large" class.
+   *
+   * @param {number} [breakpoints.xlarge]
+   *        The maximum width for the "vjs-layout-x-large" class.
+   *
+   * @param {number} [breakpoints.huge]
+   *        The maximum width for the "vjs-layout-huge" class.
+   */
+  setBreakpoints(breakpoints = {}) {
+    this.breakpoint_ = '';
+    this.breakpoints_ = assign({}, DEFAULT_BREAKPOINTS, breakpoints);
+
+    // When breakpoint definitions change, we need to update the currently
+    // selected breakpoint.
+    this.updateCurrentBreakpoint_();
+  }
 
-    this.breakpoints_ = assign({}, DEFAULT_BREAKPOINTS);
+  /**
+   * Set a flag indicating whether or not this player should adjust its UI
+   * based on its dimensions.
+   *
+   * @param  {boolean} value
+   *         Should be `true` if the player should adjust its UI based on its
+   *         dimensions; otherwise, should be `false`.
+   */
+  setResponsive(value) {
```

Comment:
why `setResponsive`/`isRespnsive` pair? Most of our other methods are getters/setters:
```js
player.responsive();
// false
player.responsive(true);
// true and responsive mode is now enabled
```

Reply:
Yeah, that's true. I thought we wanted to move away from that, though. I guess it makes sense to be more internally consistent for now.